internal/runners/config/key.go

```diff
@@ -10,7 +10,7 @@ type Key string
 
 // Set implements the captain ArgMarshaler interface
 func (k *Key) Set(v string) error {
-	regex := regexp.MustCompile(`^[A-Za-z0-9\.]+$`)
+	regex := regexp.MustCompile(`^[A-Za-z0-9]+(\.[A-Za-z0-9_]+)*$`)
```

Comment:
Nit: you don't need a group here (you can drop the brackets) and the wildcard, ie. `^[A-Za-z0-9]+\.[A-Za-z0-9_]+$`.

I'm pretty sure that would match all the same things.

Reply:
Sorry for butting in, but I disagree. The current solution matches an alphanumeric name followed by 0 or more alphanumeric+underscore names (separated by '.'). The proposed solution would match an alphanumeric name followed by a '.' followed by single alphanumeric+underscore name. The '.' and additional name would be mandatory.